packages/config/src/VitessceAutoConfig.js

```diff
@@ -0,0 +1,380 @@
+
+  async composeFileConfig() {
+    return {
+      fileType: this.fileType,
+      type: 'raster',
+      url: this.fileUrl,
+    };
+  }
+}
+
+class AnndataZarrAutoConfig extends AbstractAutoConfig {
+  constructor(fileUrl) {
+    super();
+    this.fileUrl = fileUrl;
+    this.fileType = FileType.ANNDATA_ZARR;
+    this.fileName = fileUrl.split('/').at(-1);
+    this.metadataSummary = {};
+  }
+
+  async composeFileConfig() {
+    this.metadataSummary = await this.setMetadataSummary();
+
+    const options = {
+      obsEmbedding: [],
+      obsFeatureMatrix: {
+        path: 'X',
+      },
+    };
+
+    this.metadataSummary.obsm.forEach((key) => {
+      if (key.toLowerCase().includes(('obsm/x_segmentations'))) {
+        options.obsSegmentations = { path: key };
+      }
+      if (key.toLowerCase().includes(('obsm/x_spatial'))) {
+        options.obsLocations = { path: key };
+      }
+      if (key.toLowerCase().includes('obsm/x_umap')) {
+        options.obsEmbedding.push({ path: key, embeddingType: 'UMAP' });
+      }
+      if (key.toLowerCase().includes('obsm/x_tsne')) {
+        options.obsEmbedding.push({ path: key, embeddingType: 't-SNE' });
+      }
+      if (key.toLowerCase().includes('obsm/x_pca')) {
+        options.obsEmbedding.push({ path: key, embeddingType: 'PCA' });
+      }
+    });
+
+    this.metadataSummary.obs.forEach((key) => {
+      if (key.toLowerCase().includes('cluster') || key.toLowerCase().includes('cell_type')) {
+        if (!('obsSets' in options)) {
+          options.obsSets = [
+            {
+              name: 'Cell Type',
+              path: [key],
+            },
+          ];
+        } else {
+          options.obsSets[0].path.push(key);
+        }
+      }
+    });
+
+    return {
+      options,
+      fileType: this.fileType,
+      url: this.fileUrl,
+      coordinationValues: {
+        obsType: 'cell',
+        featureType: 'gene',
+        featureValueType: 'expression',
+      },
+    };
+  }
+
+  async composeViewsConfig() {
+    this.metadataSummary = await this.setMetadataSummary();
+
+    const views = [];
+
+    const hasCellSetData = this.metadataSummary.obs
+      .filter(key => key.toLowerCase().includes('cluster') || key.toLowerCase().includes('cell_type'));
+
+    if (hasCellSetData.length > 0) {
+      views.push(['obsSets']);
+    }
+
+    this.metadataSummary.obsm.forEach((key) => {
+      if (key.toLowerCase().includes('obsm/x_umap')) {
+        views.push(['scatterplot', { mapping: 'UMAP' }]);
+      }
+      if (key.toLowerCase().includes('obsm/x_tsne')) {
+        views.push(['scatterplot', { mapping: 't-SNE' }]);
+      }
+      if (key.toLowerCase().includes('obsm/x_pca')) {
+        views.push(['scatterplot', { mapping: 'PCA' }]);
+      }
+      if (key.toLowerCase().includes(('obsm/x_segmentations'))) {
+        views.push(['layerController']);
+      }
+      if (key.toLowerCase().includes(('obsm/x_spatial'))) {
+        views.push(['spatial']);
+      }
+    });
```

Comment:
@ilan-gold what are your concerns about using `.zmetadata`? Is it that you want to avoid having the extra requirement that people have the `.zmetadata` file present?

Reply:
Exactly - here, `.zmetadata` enables not having to make for requests to check for these things but in theory, we could check for them by just making requests.  This isn't ideal, but would broaden the user-base a bit to people who have pre-existing zarr stores.  Not sure if it's worth it though.

Reply:
Each approach has its tradeoffs for sure. I like the approach of using the `.zmetadata` file and I decided to go for it, because aside from the multiple requests, it is cleaner in general -- we have everything in that file and it is our single source of truth. It feels like there is less room for ambiguity when using that file rather than having to check different `.zattrs` files and folders, it should also be easier to debug and maintain in the future.

Now the point that we require users to do one more action is a very valid one and I was wondering if the tradeoff is worth it or not. Here is my thinking. If the user managed to convert the file to a valid `Anndata-zarr` format, then they should have the skills required to run the `consolidate_metadata` function and generate the `.zmetadata` file (it is just one more `zarr` function). Weighing in the prospect of doing multiple requests + working with ambiguity vs user having to run a function they most likely will be comfortable with, I thought it is worth making that tradeoff.

Also, users already had to have programming skills to generate the `view config` themselves -- which requires working with json files and a JS/Python/R cli. So again, running a `zarr` function should not be too big of an ask.

Seeing from experience, if there is a reasonable way to avoid ambiguity, we should take it, because the costs of ambiguity are always high. It seems to me that asking the user to have the `.zmetadata` file in the folder is reasonable, given the profile of our users.

@ilan-gold, @keller-mark, I will be happy to chat more about this, let me know! Of course, you know the Vitessce users much better than me and there can be something that I am missing/over-assuming here.

Reply:
I did discuss with @ivababukova that we could rely on having `.zmetadata` and we could add information about running Zarr's `consolidate_metadata` to the docs, so I take the blame here.

You have a good point that would not be necessary to have `.zmetadata` if we can make HTTP requests to check for a small set of predefined `obsm` keys. I don't think this should necessarily block this PR though.

Reply:
Hey @ilan-gold we did discuss this with Nils and Mark during the development meeting and the agreement is that we will keep the version that uses `.zmetadata` in the current implementation, we will update the user docs to point to the "consolidate_metadata" function. In the future, we can also implement support for `Anndata-Zarr` file types without the need for the `.zmetadata` file to be present, but that is not needed at the moment. We agreed to track this as a separate issue. I will create that issue shortly and paste a link in this comments thread.

Reply:
Here is a link to the issue: https://github.com/vitessce/vitessce/issues/1459